### DISCUSSION TITLE
How to return different type from query than resolver is meant for?

### DISCUSSION THREAD

**Discussion by mattoni** [MichalLytek/type-graphql#901] [Q&A]
Hello,

Say I want to get the author from a post, but the post only has an author ID (simple).

```
@ObjectType()
export class Post {
  @Field(type => Author) 
  author: Author
}

@ObjectType()
export class Author {
  @Field() 
  id: string

  @Field()
  name: string
}
```

But, from the database we only have `authorId`.

```
@Resolver((of) => Post)
export class PostResolver {

  @Query((returns) => Post)
    async getPost() {
      return { 
        id: 'myPostId',
        authorId: 'myAuthorId'
      }
    }

   @FieldResolver()
    async author(@Root() post: Post) {
     // post doesn't have authorId here
   }

}
```

What's the idiomatic way to structure this with TypeGraphQL? Essentially, I want my query resolver to return something different than the type it's resolving for, and let the field resolvers work it out. 

Thanks!

**Comment by MichalLytek** [comment]
```ts
@ObjectType()
class Rate {
  @Field(type => Int)
  value: number;

  @Field()
  date: Date;

  user: User;
}
```

> Also the user property doesn't have a `@Field()` decorator - this way we can hide some properties of our data model. In this case, we need to store the user field of the Rate object to the database in order to prevent multiple rates, but we don't want to make it publicly accessible.



https://typegraphql.com/docs/types-and-fields.html
